src/java/org/apache/cassandra/cql3/statements/schema/CreateIndexStatement.java

```diff
@@ -115,11 +115,11 @@ public Keyspaces apply(Keyspaces schema)
                     throw ire("Duplicate column '%s' in index target list", target.column);
         }
 
-        indexTargets.forEach(t -> validateIndexTarget(table, t));
+        IndexMetadata.Kind kind = attrs.isCustom ? IndexMetadata.Kind.CUSTOM : IndexMetadata.Kind.COMPOSITES;
```

Comment:
@djatnieks I'm tagging you for review of this commit please. This contains the changes to the Cassandra Index API to allow multiple index expressions to be used in a `SELECT` query without needing `ALLOW FILTERING`. The original work for this was done in DB-2319.